File: cpp/Gisketch/Aottg2UI/Search/GisketchSearchMatch.c
// Type: Gisketch.Aottg2UI.Search.GisketchSearchMatch
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// C# structure: source/csharp/Gisketch.Aottg2UI.Runtime/Gisketch.Aottg2UI.Search/GisketchSearchMatch.cs
// Prior source: NEW in this update
// --------------------------------

// Gisketch.Aottg2UI.Search.GisketchSearchMatch$$.ctor
// il2cpp: void Gisketch_Aottg2UI_Search_GisketchSearchMatch___ctor (Gisketch_Aottg2UI_Search_GisketchSearchMatch_o __this, int32_t score, int32_t displayStart, int32_t displayLength, System_String_o* hiddenText, System_Int32_array* displayIndices, const MethodInfo* method);
// 0x3ae1780

void Gisketch_Aottg2UI_Search_GisketchSearchMatch___ctor
               (Gisketch_Aottg2UI_Search_GisketchSearchMatch_o __this,int32_t score,
               int32_t displayStart,int32_t displayLength,System_String_o *hiddenText,
               System_Int32_array *displayIndices,MethodInfo *method)

{
  undefined4 in_register_0000003c;
  int32_t *piVar1;
  
  piVar1 = (int32_t *)CONCAT44(in_register_0000003c,score);
  *piVar1 = displayStart;
  piVar1[1] = displayLength;
  piVar1[2] = (int32_t)hiddenText;
  if (displayIndices == (System_Int32_array *)0x0) {
    displayIndices = (System_Int32_array *)**(undefined8 **)(DAT_057110b0 + 0xb8);
  }
  *(System_Int32_array **)(piVar1 + 4) = displayIndices;
  il2cpp_runtime_glue(piVar1 + 4,displayIndices);
  *(MethodInfo **)(piVar1 + 6) = method;
  il2cpp_runtime_glue(piVar1 + 6,method);
  return;
}

File: csharp_body_deref/Gisketch.Aottg2UI.Runtime/Gisketch.Aottg2UI.Search/GisketchSearchMatch.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Gisketch.Aottg2UI.Search.GisketchSearchMatch
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Gisketch/Aottg2UI/Search/GisketchSearchMatch.c
// Prior real C# source: none (type is newer than the last public source snapshot)
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using System.Runtime.InteropServices;
using Il2CppDummyDll;

namespace Gisketch.Aottg2UI.Search;

[Token(Token = "0x200002F")]
public readonly struct GisketchSearchMatch
{
	[Token(Token = "0x40000E5")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x0")]
	public readonly int Score;

	[Token(Token = "0x40000E6")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x4")]
	public readonly int DisplayStart;

	[Token(Token = "0x40000E7")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x8")]
	public readonly int DisplayLength;

	[Token(Token = "0x40000E8")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x10")]
	public readonly string HiddenText;

	[Token(Token = "0x40000E9")]
	[Il2CppDummyDll.FieldOffset(Offset = "0x18")]
	public readonly int[] DisplayIndices;

	[Token(Token = "0x6000182")]
	[Address(RVA = "0x3A4BE00", Offset = "0x3A4BE00", VA = "0x3A4BE00")]
	public GisketchSearchMatch(int score, int displayStart, int displayLength, string hiddenText, [Optional] int[] displayIndices)
	{
	}
}
